docs/interactive-mode.md

```diff
@@ -0,0 +1,179 @@
+> Note: This mode does not provide support for bulk resource export or import.
+
+### Running the tool in the interactive mode
+#### Tool Initialization
+1. Setup the tool and the IS, following the steps in the [How to run the tool ](../README.md##How to run the tool ) section.
+2. Run the following command to initialize the tool by providing the Identity server details and client ID/secret of the app you created.
+```
+iamctl init
+```
+Provide the details as prompted by the tool.
+```
+:~$ iamctl init
+  ___      _      __  __            ____   _____   _     
+ |_ _|    / \    |  \/  |          / ___| |_   _| | |    
+  | |    / _ \   | |\/| |  _____  | |       | |   | |    
+  | |   / ___ \  | |  | | |_____| | |___    | |   | |___ 
+ |___| /_/   \_\ |_|  |_|          \____|   |_|   |_____|
+      
+? Enter IAM URL [<schema>://<host>]: https://localhost:9443                                                   
+? Enter clientID: *******
+? Enter clientSecret: *******
+? Enter Tenant domain: carbon.super
+```
+Run the following command to provide admin user credentials.
+```
+iamctl serverConfiguration [flags]
+```
+
+Flags:
+```
+  -h, --help              help for serverConfiguration
+  -p, --password string   enter your password
+  -s, --server string     set server domain
+  -u, --username string   enter your username
+```
+example:-
+```
+iamctl serverConfiguration -h                                           //help for serverConfiguration
+iamctl serverConfiguration -s=https://localhost:9443 -u=admin -p=*****  //to complete the authorization
+```
+Set admin user credentials by entering inputs in an interactive way.
+```
+iamctl serverConfiguration
+```
+example:-
+```
+~$ iamctl serverConfiguration 
+? Enter IAM URL [<schema>://<host>]: https://localhost:9443
+? Enter Username: admin
+? Enter Password: *****
+```
+
+### Application Related Commands
+**Add application**
+```
+iamctl application [commands]
+iamctl application     add      [flags]
+```
+
+Flags:
+ ```
+   -c, --callbackURl string    callbackURL  of SP - **for oauth application
+   -d, --description string    description of SP - **for basic application
+   -h, --help                  help for add
+   -n, --name string           name of service provider - **compulsory
+   -p, --password string       Password for Identity Server
+   -s, --serverDomain string   server Domain
+   -t, --type string           Enter application type (default "oauth")
+   -u, --userName string       Username for Identity Server
+ ```
+Users have freedom to set flags and values according to their choices.
+
+This ```-t, --type string           Enter application type (default "oauth")``` flag  is not mandatory. If user wants to create basic application, then should declare ```-t=basic```. Otherwise will create the oauth application as default type.
```

Comment:
```suggestion
This ```-t, --type string           Enter application type (default "oauth")``` flag  is not mandatory. If the user wants to create a basic application, it is necessary to declare ```-t=basic```. Otherwise, the OAuth application will be created as the default type.
```